teps/0059-skip-guarded-task-only.md

```diff
@@ -95,6 +107,28 @@ updates.
 
 This TEP addresses skipping strategies to give users the flexibility to skip a single guarded `Task` only and unblock execution of its dependent `Tasks`.
 
+Today, we support specifying a list of `WhenExpressions` through the `when` field as such:
+
+```yaml
+when:
+  - input: 'foo'
+    operator: in
+    values: [ 'bar' ]
+```
+
+We propose changing the `when` field from a list to a dictionary and adding `scope` and `expressions` fields under the `when` field.
+- The `scope` field would be used to specify whether the `WhenExpressions` guard the `Task` only or the whole `Branch`.
+- The `expressions` field would be used to specify the list of `WhenExpressions`, each of which has `input`, `operator` and `values` fields.
+
+```yaml
+when:
+  scope: Task / Branch
+  expressions:
+    - input: 'foo'
+      operator: in
+      values: [ 'bar' ]
```

Comment:
Any reason for having the `scope` at root level and not associated to individual expressions?
This looks fine to me, it keeps things simpler - but I was wondering if there had been any special reasoning about that already.

I think one of the reasons that is not straight forward to express the behaviour of conditions, is because we don't explicitly express / name branches in the pipeline, they're derived from dependencies. Having the scope attached to a pipeline task seems like a good compromise.

Reply:
> Any reason for having the `scope` at root level and not associated to individual expressions?
> This looks fine to me, it keeps things simpler - but I was wondering if there had been any special reasoning about that already.

interesting 🤔 

we haven't considered this idea yet, will think more about it

Reply:
After watching the API WG recording, I wanted to clarify I didn't mean to block the TEP on this question :)
It was just not clear to me why the scope was always attached to the list of conditions.

I've been thinking about a few use cases and as long as there is only one branch under the guarded task, it's possible to do translate this:
```
A -> B (Cond1@task, Cond2@branch) -> C -> D
```
into:
```
A -> B (Cond1@task) -> C (Cond2@branch) -> D
```

If there is more than one branch after B, it is still possible to do the translation, but `Cond2` needs to be repeated on the top of each branch. This feels like a reasonable compromise in favour of having a more simple syntax and implementation. Also the proposed syntax still allows for future developments and adding more granularity in future if we decide to.

Reply:
wasn't blocked on the question, it was more that i wanted us to consider how the proposed solution would extend to support this scenario in the future so was asking if folks in the API working group had use cases to help in this design

thanks for the follow up @afrittoli!